sklego/meta/outlier_classifier.py

```diff
@@ -31,6 +32,11 @@ def fit(self, X, y=None):
             raise ValueError("Passed model does not detect outliers!")
         self.estimator_ = self.model.fit(X, y)
         self.classes_ = np.array([0, 1])
+
+        # fit sigmoid function for `predict_proba`
+        decision_function_scores = self.estimator_.decision_function(X)
```

Comment:
I just checked, *technically* there [is an outlier detection mechanism in scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.LocalOutlierFactor.html) that doesn't support `decision_function`. That it's a bit of an oddball since it only supports `fit_predict`. Theoretically we might want to check/confirm that the decision function exists, but practically it's not something we should expect.

Reply:
if `novelty=True` it also works with the LocalOutlierFactor. I can add an assertion for that and raise an error if this setting is not chosen.

Reply:
Checking for a setting on a specific outlier detection model might be too "specific". Now that I think about it. I suppose wrapping it with a `try`/`except` such that a meaningful error message is given is sufficient.

Reply:
we could also check if a `decision_function` is available, then it's not specific for the LocalOutlierFactor but also not so wide as using a try/except. I'll commit it, let me know what you think about it.